src/mesh/FloodingRouter.cpp

```diff
@@ -68,6 +68,13 @@ bool FloodingRouter::isRebroadcaster()
 void FloodingRouter::perhapsRebroadcast(const meshtastic_MeshPacket *p)
 {
     if (!isToUs(p) && (p->hop_limit > 0) && !isFromUs(p)) {
+        // If this packet arrived via UDP and is still encrypted, skip rebroadcast.
```

Comment:
This causes packets to not be rebroadcasted when they use PKI for other nodes or channels you don't have. I believe this can be safely removed, as the Router will try to decrypt before calling the FloodingRouter; it will not end up here twice.